watchtower/wtclient/client.go

```diff
@@ -596,6 +670,19 @@ func (c *TowerClient) backupDispatcher() {
 				// additional sessions.
 				goto awaitSession
 
+			// A new tower has been added, so we'll update our
+			// in-memory state with an entry for it in our candidate
+			// iterator and consider its corresponding sessions as
+			// new candidates.
+			case t := <-c.newTowers:
+				c.handleNewTower(t)
+
+			// A tower has been removed, so we'll remove its
+			// entry within the candidate iterator and its
+			// corresponding sessions from our in-memory state.
+			case t := <-c.staleTowers:
+				c.handleStaleTower(t)
```

Comment:
prevent removing towers while already negotiating? coudl be a bit racy with the async dispatcher

Reply:
Done.